src/app/qgisapp.cpp

```diff
@@ -2618,34 +2618,58 @@ void QgisApp::dataSourceManager( const QString &pageName )
     mDataSourceManagerDialog = new QgsDataSourceManagerDialog( mBrowserModel, this, mapCanvas() );
     connect( this, &QgisApp::connectionsChanged, mDataSourceManagerDialog, &QgsDataSourceManagerDialog::refresh );
     connect( mDataSourceManagerDialog, &QgsDataSourceManagerDialog::connectionsChanged, this, &QgisApp::connectionsChanged );
-    connect( mDataSourceManagerDialog, &QgsDataSourceManagerDialog::addRasterLayer,
-             this, [this]( const QString & uri, const QString & baseName, const QString & providerKey )
+
+
+    connect( mDataSourceManagerDialog, &QgsDataSourceManagerDialog::addLayer,
+             this, [this]( Qgis::LayerType type, const QString & uri, const QString & baseName, const QString & providerKey )
     {
-      addRasterLayer( uri, baseName, providerKey );
+      switch ( type )
+      {
+        case Qgis::LayerType::Raster:
+          addRasterLayer( uri, baseName, providerKey );
+          break;
+
+        case Qgis::LayerType::Vector:
+        {
+          if ( QgsVectorLayer *layer = addVectorLayer( uri, baseName, providerKey ) )
+            QgsAppLayerHandling::postProcessAddedLayers( {layer} );
```

Comment:
Do we have an established rule on `( { layer } )` vs `( {layer} )`?

Reply:
I don't think so -- it's whatever random formatting astyle decides applies in each situation.